src/ZODB/_compat.py

```diff
@@ -61,14 +72,58 @@ def loads(s):
     FILESTORAGE_MAGIC = b"FS30"
 
 
-# XXX: consistent spelling of inst_persistent_id/persistent_id?
-#      e.g. StorageTestBase and probably elsewhere
+def PersistentPickler(persistent_id, *args, **kwargs):
+    """
+    Returns a :class:`Pickler` that will use the given ``persistent_id``
+    to get persistent IDs. The remainder of the arguments are passed to the
+    Pickler itself.
+
+    This covers the differences between Python 2 and 3 and PyPy/zodbpickle.
+    """
+    p = Pickler(*args, **kwargs)
+    if sys.version_info[0] < 3:
+        p.inst_persistent_id = persistent_id
+        # PyPy uses a python implementation of cPickle in both Python 2
+        # and Python 3. We can't really detect inst_persistent_id as its
+        # a magic attribute that's not readable, but it doesn't hurt to
+        # simply always assign to persistent_id also
+        p.persistent_id = persistent_id
+    else:
+        p.persistent_id = persistent_id
+    return p
+
+def PersistentUnpickler(find_global, load_persistent, *args, **kwargs):
+    """
+    Returns a :class:`Unpickler` that will use the given `find_global` function
+    to locate classes, and the given `load_persistent` function to load
+    objects from a persistent id.
+
+    This covers the differences between Python 2 and 3 and PyPy/zodbpickle.
+    """
+    unpickler = Unpickler(*args, **kwargs)
+    if find_global is not None:
+        unpickler.find_global = find_global
+        try:
+            unpickler.find_class = find_global # PyPy, zodbpickle, the non-c-accelerated version
+        except AttributeError:
+            pass
+    if load_persistent is not None:
+        unpickler.persistent_load = load_persistent
+
+    return unpickler
 
 
 try:
     # Python 2.x
-    # XXX: why not just import BytesIO from io?
-    from cStringIO import StringIO as BytesIO
+    if IS_JYTHON:
+        # Jython 2.7rc2 cStringIO.StringIO class has a bug
+        # resulting in StringIndexOutOfBoundExceptions
+        # when repeatedly writing and then seeking back to 0
+        # http://bugs.jython.org/issue2324
+        from io import BytesIO
+    else:
+        # XXX: why not just import BytesIO from io?
+        from cStringIO import StringIO as BytesIO
```

Comment:
Given that the fix for http://bugs.jython.org/issue2324 [landed in Jython 2.7.0](https://hg.python.org/jython/file/412a8f9445f7/NEWS#l17), can we just back this part out??

Also, I'm still not clear why we are not just using `io.BytesIO` everywhere.